shell/platform/windows/win32_window.cc

```diff
@@ -109,10 +110,70 @@ void Win32Window::TrackMouseLeaveEvent(HWND hwnd) {
   }
 }
 
+void Win32Window::OnImeSetContext(UINT const message,
+                                  WPARAM const wparam,
+                                  LPARAM const lparam) {
+  if (wparam != 0) {
+    text_input_manager_.CreateImeWindow(window_handle_);
+  }
+}
+
+void Win32Window::OnImeStartComposition(UINT const message,
+                                        WPARAM const wparam,
+                                        LPARAM const lparam) {
+  text_input_manager_.CreateImeWindow(window_handle_);
+  OnComposeBegin();
+}
+
+void Win32Window::OnImeComposition(UINT const message,
+                                   WPARAM const wparam,
+                                   LPARAM const lparam) {
+  // Update the IME window position.
+  text_input_manager_.UpdateImeWindow(window_handle_);
+
+  if (lparam & GCS_COMPSTR) {
+    // Read the in-progress composing string.
+    long pos = text_input_manager_.GetComposingCursorPosition(window_handle_);
+    std::u16string text{};
```

Comment:
Why `text{}` rather than `text`? We don't usually use that style.

Reply:
Oops -- thanks for spotting. Had been poking at a bug and apparently deleted the string but missed the brackets (then copied and pasted it apparently). Done.